R/classes.R

```diff
@@ -1,218 +1,290 @@
-  print(round(object$`delta h ACAI SFI-PFI (deletion)`[c(item_set), ,
-                                                       drop = FALSE], 3))
-  cat(paste0("\n", stars, "\nAdverse Impact (AI) ratio for item subsets by 
-             invariance condition:\n", stars, "\n"))
-  print(round(object$AI[c(1, item_set + 1), , drop = FALSE], 3))
-  cat(paste0("\n", stars, "\nCOMPARING CAI FOR REFERENCE AND (EXPECTED) FOCAL 
-             GROUPS\n", stars))
-  cat("\nDiscrepancy between CAI of reference vs. Efocal groups:\n")
-  print(round(object$`h CAI Ref-EF`[c(1, item_set + 1), , drop = FALSE], 3))
-  cat(paste0(dashes, "\nImpact of deleting an item on the discrepancy between 
-             CAI of reference \nvs. Efocal groups:\n"))
-  print(round(object$`delta h CAI Ref-EF (deletion)`[c(item_set), ,
-                                                     drop = FALSE], 3))
-  cat(paste0("\n", stars, "\nCOMPARING CAI UNDER STRICT AND PARTIAL FACTORIAL
-             INVARIANCE\n", stars))
-  cat("\nDiscrepancy between CAI under SFI vs. PFI for the reference group:\n")
-  print(round(object$`h CAI SFI-PFI`[[1]][c(1, item_set + 1), , drop = FALSE],
-              3))
-  cat("\nDiscrepancy between CAI under SFI vs. PFI for the focal group:\n")
-  print(round(object$`h CAI SFI-PFI`[[2]][c(1, item_set + 1), , drop = FALSE],
-              3))
-  cat(paste0(dashes, "\nImpact of deleting an item on the discrepancy between
-             CAI under SFI \nvs. PFI for the reference group:\n"))
-  print(round(object$`delta h SFI-PFI (deletion)`[[1]][c(item_set), ,
-                                                       drop = FALSE], 3))
-  cat("\nImpact of deleting an item on the discrepancy between CAI under SFI 
-      \nvs. PFI for the focal group:\n")
-  print(round(object$`delta h SFI-PFI (deletion)`[[2]][c(item_set), ,
-                                                       drop = FALSE], 3))
-  print(object$`h SFI-PFI by groups`$reference)
-  print(object$`h SFI-PFI by groups`$focal)
-  print(object$PartInv$strict)
-  print(object$PartInv$partial)
+.print_idel_tbl <- function(x, digits, cols = 5:8,
+                            first_is_ref = FALSE, ...) {
+  if (is.list(x) && length(dim(x[[1]])) >= 2) {
+    lapply(seq_along(x), function(i) {
+      if (i == 1 && first_is_ref) {
+        glab <- "Reference"
+      } else {
+        glab <- "Focal"
+      }
+      cat(glab, " group: ", names(x[i]), "\n", sep = "")
+      .print_idel_tbl(x[[i]], digits = digits, cols = cols, ...)
+    })
+  } else {
+    df <- as.data.frame(x[, cols, drop = FALSE])
+    
+    # round and format numeric columns, control # of digits after the decimal
+    num_cols <- sapply(df, is.numeric)
+    if (any(num_cols)) { 
+      df[num_cols] <- lapply(df[num_cols], function(v)
+        # format = "f" to prevent scientific notation, drop0trailing = F to keep 0s
+        formatC(v, digits = digits, format = "f", drop0trailing = FALSE))
+    }
+    
+    print(df, right = TRUE,  ...)
```

Comment:
```suggestion
    print(df, ...)
```

Personally I'm not a big friend of using behaviors that are quite different from the base R print method, as there may lead to unexpected issues when users set a global option for the print method and seeing the option not being applied in specific functions. But I respect your choice

Reply:
Thank you for your note. I removed 'right = TRUE' and now the default version is used.